src/main/java/qupath/ext/instanseg/core/InstanSegModel.java

```diff
@@ -251,8 +258,64 @@ public Optional<Integer> getNumChannels() {
         return getModel().flatMap(model -> Optional.of(extractChannelNum(model)));
     }
 
-    private static int extractChannelNum(BioimageIoSpec.BioimageIoModel model) {
-        int ind = model.getInputs().getFirst().getAxes().toLowerCase().indexOf("c");
+    /**
+     * Try to check the output tensors from the model spec.
+     * @return The output tensors if the model is downloaded, otherwise empty.
+     */
+    public Optional<List<OutputTensor>> getOutputs() {
+        return getModel().flatMap(model -> Optional.ofNullable(model.getOutputs()));
+    }
+
+    /**
+     * Try to check the output classes from the model spec.
+     * @return The output classes if the model is downloaded, and it's present, otherwise empty.
+     */
+    public List<String> getClasses() {
+        var config = model.getConfig().getOrDefault("qupath", null);
+        if (config instanceof Map configMap) {
+            List<String> classes = new ArrayList<>();
+            var el = configMap.get("classes");
+            if (el != null && el instanceof List elList) {
+                for (var t: elList) {
+                    classes.add(t.toString());
+                }
+            }
+            return classes;
+        }
+        return List.of();
+    }
+
+    public enum OutputType {
+        // "instance segmentation" "cell embeddings" "cell classes" "cell probabilities" "semantic segmentation"
+        INSTANCE_SEGMENTATION("instance_segmentation"),
+        DETECTION_EMBEDDINGS("detection_embeddings"),
+        DETECTION_LOGITS("detection_logits"),
+        DETECTION_CLASSES("detection_classes"),
+        SEMANTIC_SEGMENTATION("semantic_segmentation");
+
+        private final String type;
+        OutputType(String type) {
```

Comment:
Please include a blank line to ensure fields are clearly separated from constructors.
Also make private if it doesn't need to be package private.

Reply:
IDE linting tells me private doesn't make any odds here